server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSStartbrowseOptionsCheckUtility.java

```diff
@@ -0,0 +1,111 @@
+        {
+            put(CICSLexer.ACTIVITYID, ErrorSeverity.ERROR);
+            put(CICSLexer.PROCESS, ErrorSeverity.ERROR);
+            put(CICSLexer.PROCESSTYPE, ErrorSeverity.ERROR);
+            put(CICSLexer.ACTIVITY, ErrorSeverity.WARNING);
+            put(CICSLexer.BROWSETOKEN, ErrorSeverity.ERROR);
+            put(CICSLexer.CONTAINER, ErrorSeverity.WARNING);
+            put(CICSLexer.CHANNEL, ErrorSeverity.WARNING);
+            put(CICSLexer.EVENT, ErrorSeverity.WARNING);
+            put(CICSLexer.TIMER, ErrorSeverity.WARNING);
+            put(CICSLexer.RETCODE, ErrorSeverity.ERROR);
+        }
+    };
+
+    /**
+     * Checks CICS Startbrowse rules for required and invalid options
+     * @param dialectProcessingContext
+     * @param errors
+     */
+    public CICSStartbrowseOptionsCheckUtility(
+        DialectProcessingContext dialectProcessingContext,
+        List<SyntaxError> errors) {
+      super(dialectProcessingContext, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS STARTBROWSE rule options
+     *
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        if (ctx.getRuleIndex() == CICSParser.RULE_cics_startbrowse_body) {
+            checkBody((CICSParser.Cics_startbrowse_bodyContext) ctx);
+        }
+    }
+
+    private void checkBody(CICSParser.Cics_startbrowse_bodyContext ctx) {
+        if (!ctx.ACTIVITY().isEmpty()) {
+            checkPrerequisiteIsMet(ctx.cics_startbrowse_processWithValue_subrule(), ctx.PROCESSTYPE(), ctx, "PROCESSTYPE without PROCESS");
+            checkHasIllegalOptions(ctx.PROCESS(), "PROCESS, in this context, requires a value");
+            checkMutuallyExclusiveOptions("ACTIVITYID or PROCESS", ctx.ACTIVITYID(), ctx.PROCESS());
+        } else if (!ctx.CONTAINER().isEmpty()) {
+            checkPrerequisiteIsMet(ctx.cics_startbrowse_processWithValue_subrule(), ctx.PROCESSTYPE(), ctx, "PROCESSTYPE without PROCESS");
+            checkHasIllegalOptions(ctx.PROCESS(), "PROCESS, in this context, requires a value");
+            checkMutuallyExclusiveOptions("ACTIVITYID, PROCESS or CHANNEL", ctx.ACTIVITYID(), ctx.PROCESS(), ctx.cics_startbrowse_processWithValue_subrule(), ctx.CHANNEL());
+        } else if (!ctx.EVENT().isEmpty()) {
+            checkHasIllegalOptions(ctx.CHANNEL(), "CHANNEL");
+            checkHasIllegalOptions(ctx.PROCESS(), "PROCESS");
+            checkHasIllegalOptions(ctx.cics_startbrowse_processWithValue_subrule(), "PROCESS");
+            checkHasIllegalOptions(ctx.PROCESSTYPE(), "PROCESSTYPE");
+        } else if (!ctx.PROCESS().isEmpty()) {
+            checkHasMandatoryOptions(ctx.PROCESSTYPE(), ctx, "PROCESSTYPE");
+            checkHasIllegalOptions(ctx.cics_startbrowse_processWithValue_subrule(), "PROCESS, in this context, cannot have a value");
```

Comment:
ACTIVITYID
and
ACTIVITY 
CONTAINER 
EVENT 
CHANNEL
TIMER

are illegal here.